runc.go

```diff
@@ -174,14 +193,18 @@ func (r *Runc) Create(context context.Context, id, bundle string, opts *CreateOp
 	cmd.ExtraFiles = opts.ExtraFiles
 
 	if cmd.Stdout == nil && cmd.Stderr == nil {
-		data, err := cmdOutput(cmd, true, nil)
+		data, err := r.cmdOutput(cmd, true, nil)
 		defer putBuf(data)
 		if err != nil {
 			return fmt.Errorf("%s: %s", err, data.String())
 		}
 		return nil
 	}
-	ec, err := Monitor.Start(cmd)
+	if r.PdeathSignal != 0 {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+	}
+	ec, err := r.startCommand(cmd)
```

Comment:
(multiple occurrences) Since there's now locking in `Monitor.StartLocked` and the `r.startCommand` reads `r.PdeathSignal` to switch, I don't think you need this locking here.

```suggestion
	ec, err := r.startCommand(cmd)
```

Reply:
Oops, missed that. Fixed now.